lib/src/main/java/graphql/nadel/engine/transform/partition/NadelPartitionTransformHook.kt

```diff
@@ -0,0 +1,54 @@
+package graphql.nadel.engine.transform.partition
+
+import graphql.language.ScalarValue
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecutionHydrationDetails
+import graphql.nadel.engine.NadelExecutionContext
+import graphql.nadel.engine.NadelServiceExecutionContext
+import graphql.nadel.engine.blueprint.NadelOverallExecutionBlueprint
+import graphql.normalized.ExecutableNormalizedField
+import graphql.schema.GraphQLInputValueDefinition
+
+/**
+ * This hook allows you to provide custom partitioning logic for fields.
+ */
+interface NadelPartitionTransformHook {
+
+    /**
+     * This hook allows you to provide a custom context based on the characteristics of the field being partitioned.
+     *
+     * This context will be passed to the partition key extractor function so you can use it to determine the partition key.
+     *
+     * You can use this to check custom directives on the field, or any other details of the field definition.
+     *
+     * If `null` is returned, the field will not be partitioned.
+     *
+     */
+    fun getFieldPartitionContext(
+        executionContext: NadelExecutionContext,
+        serviceExecutionContext: NadelServiceExecutionContext,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        services: Map<String, Service>,
+        service: Service,
+        overallField: ExecutableNormalizedField,
+        hydrationDetails: ServiceExecutionHydrationDetails?,
+    ): Any? {
```

Comment:
Think we should make a parent class like `NadelFieldPartitionContext` so we don't have to pass around `Any`

e.g.

```
abstract class NadelFieldPartitionContext
```

Similar to how `NadelServiceExecutionContext` does it.

Reply:
Love it. It will make the code much more readable.